ci-build/test.sh

```diff
@@ -2,35 +2,42 @@
 # Copyright 2016 TiKV Project Authors
 
 set -o pipefail
+set -e
 
 panic() {
-    echo -e "$@" >&1
+    echo -e "$@" >&2
     exit 1
 }
 
-if [[ "$SKIP_FORMAT_CHECK" != "true" ]]; then
+# Move to project root
+cd "$(dirname "$0")/.."
+
+if [[ -z "$SKIP_FORMAT_CHECK" ]]; then
     make format
-    git diff-index --quiet HEAD -- || (git diff; panic "\e[35mplease make format before creating a pr!!!\e[0m")
+    git diff-index --quiet HEAD -- || (git --no-pager diff; panic "\e[35mplease 'make format' before creating a pr!!!\e[0m")
 fi
 
 trap 'kill $(jobs -p) &> /dev/null || true' EXIT
 
-export LOG_FILE=tests.log
 if [[ "$TRAVIS" = "true" ]]; then
     export RUST_TEST_THREADS=2
 fi
 export RUSTFLAGS=-Dwarnings
 
-make clippy || panic "\e[35mplease fix the errors!!!\e[0m"
+make clippy || panic "\e[35mplease fix the 'make clippy' errors!!!\e[0m"
 
-if [[ "$SKIP_TESTS" != "true" ]]; then
-    make test 2>&1 | tee tests.out
+set +e
+export LOG_FILE=tests.log
+if [[ -z "$SKIP_TESTS" ]]; then
+    ./scripts/test-all 2>&1 | tee tests.out
```

Comment:
I suggest to use `make test` instead. Using multiple entrances for the same functionality can lead to bug easily.

Reply:
I don't mind changing it to `make test`. However, that won't actually change the fact that both entry points exist.

Make is designed to help build by tracking build dependencies. In the role as build tool all building should be done through make. But we are instead using it as a convenience for running the most common commands (BTW, there are tools [designed](https://github.com/casey/just) for this). In this role there will always be the possibility to run commands directly instead of through the convenience of make.

Reply:
The line below runs `./scripts/test-all --no-run` and has an argument. The problem with using make is that the arguments must then be set as environment variables (or make variables). The problem with this approach is that there is no verification that the environment variable is properly spelled. Whereas CLI arguments are verified (or in this case even simpler they are just passed through).

So If I write `CARGO_ARGS=--no-run make test` that silently doesn't work. Nothing is telling me that I was supposed to write `EXTRA_CARGO_ARGS=`